### PR TITLE
Use background gas price estimation method in new ui.

### LINKED ISSUE #4615: NewUI Estimating high gas price
Using v4.8.0, MetaMask estimated 1gwei on old UI then 21 gwei on New UI.

We should make sure we're using the same estimation logic, since it saves users a lot of money.

This visual shows why:
https://danfinlay.github.io/eth-gas-price-visuals/

### PR DESCRIPTION
**Pull request by danjm**
Fixes #4615

### CODE REVIEW

File: ui/app/actions.js [hunk 1]
```diff
@@ -740,21 +740,27 @@ function updateGasData ({
   to,
   value,
 }) {
-  const estimatedGasPrice = estimateGasPriceFromRecentBlocks(recentBlocks)
   return (dispatch) => {
-    return Promise.all([
-      Promise.resolve(estimatedGasPrice),
-      estimateGas({
+    let gasPrice
+    return (() => new Promise((resolve, reject) => {
```
[the remark this answers is not in the document]
Comment (whymarrh) [reply ui/app/actions.js]: Laziness

File: ui/app/actions.js [hunk 2]
```diff
@@ -746,20 +745,26 @@ function updateGasData ({
 }) {
   return (dispatch) => {
     dispatch(actions.gasLoadingStarted())
-    const estimatedGasPrice = estimateGasPriceFromRecentBlocks(recentBlocks)
-    return Promise.all([
-      Promise.resolve(estimatedGasPrice),
-      estimateGas({
+    let gasPrice
+    return new Promise((resolve, reject) => {
+       background.getGasPrice((err, data) => {
```
Comment (alextsg) [inline ui/app/actions.js]: Looks like the indentation is inconsistent.

File: ui/app/actions.js [hunk 3]
```diff
@@ -746,20 +745,26 @@ function updateGasData ({
 }) {
   return (dispatch) => {
     dispatch(actions.gasLoadingStarted())
-    const estimatedGasPrice = estimateGasPriceFromRecentBlocks(recentBlocks)
-    return Promise.all([
-      Promise.resolve(estimatedGasPrice),
-      estimateGas({
+    let gasPrice
```
Comment (alextsg) [inline ui/app/actions.js]: Why is `let gasPrice` needed here?

File: ui/app/actions.js (hunk 2 above)
Comment (danjm) [reply ui/app/actions.js]: We need the value resolved by the first promise, it then needs to be passed to `estimateGas` which returns another promise, and then is needed again after `estimateGas` resolves.

File: ui/app/actions.js (hunk 3 above)
Comment (danjm) [reply ui/app/actions.js]: We need the value resolved by the first promise, it then needs to be passed to estimateGas which returns another promise, and then is needed again after estimateGas resolves.

**Review by alextsg — DISMISSED** [review]

File: ui/app/actions.js (hunk 3 above)
Comment (whymarrh) [reply ui/app/actions.js]: As an alternative, the old form doesn't require external state (i.e. using `Promise.resolve(gasPrice)` to pass the gas price through the promise chain):

```js
return new Promise((resolve, reject) => {
  background.getGasPrice((err, data) => {
    if (err !== null) return reject(err)
    return resolve(data)
  })
})
.then(estimateGasPrice => {
  return Promise.all([
    Promise.resolve(estimateGasPrice),
    estimateGas({
      estimateGasMethod: background.estimateGas,
      blockGasLimit,
      selectedAddress,
      selectedToken,
      to,
      value,
      gasPrice,
    })
  ])
})
.then(([gasPrice, gas]) => {
  dispatch(actions.setGasPrice(gasPrice))
  dispatch(actions.setGasLimit(gas))
  return calcGasTotal(gas, gasPrice)
})
```

**Review by whymarrh — APPROVED** [review]

**Pull request merged**